plugins/module_utils/common_ionos_methods.py

```diff
@@ -113,6 +113,22 @@ def get_users(api, all_users, depth=2, query_params=None):
 
     return all_users
 
+
+def get_paginated(method, depth=1, query_params=None):
```

Comment:
looks okay, but from what I understand:

(and assuming this func will remain used mostly by `get_resource_id`)

This func will still, immediately build a full list of resources at once, say `1000`.

If the resource we're looking for is the first in the list, all the other 9 API calls other than the first one are redundant

And this being said, couldn't we use like a `yield` to stop as soon as we find the one we're looking for ?